src/main/java/org/spongepowered/api/world/gen/populator/Dungeon.java

```diff
@@ -75,12 +78,50 @@ default void setAttemptsPerChunk(int attempts) {
     }
 
     /**
-     * Gets the {@link MobSpawnerData} which represents the MobSpawner which
-     * will be created within the dungeon.
-     * 
-     * @return The mob spawner data
+     * <p>Gets the {@link MobSpawnerData} which represents the MobSpawner which
+     * will be created within the dungeon.</p>
+     *
+     * <p><b>Note: </b> Only one of choices or mob spawner data
+     * will be present.</p>
+     *
+     * @return The mob spawner data, if present
+     */
+    Optional<MobSpawnerData> getMobSpawnerData();
```

Comment:
Dungeons by default randomly choose a random `EntityType` to spawn, so not always going to have the same choice of mob spawner data.